lib/recurly/risk/three-d-secure/strategy/cybersource.js

```diff
@@ -0,0 +1,101 @@
+import errors from '../../../errors';
+import Promise from 'promise';
+import ThreeDSecureStrategy from './strategy';
+import { Frame } from '../../../frame';
+
+const debug = require('debug')('recurly:risk:three-d-secure:cybersource');
+
+export default class CybersourceStrategy extends ThreeDSecureStrategy {
+  static strategyName = 'cybersource';
+  static PREFLIGHT_TIMEOUT = 30000;
+
+  static preflight ({ recurly, number, month, year, gateway_code }) {
+    const { PREFLIGHT_TIMEOUT } = CybersourceStrategy;
+    const data = {
+      gatewayType: CybersourceStrategy.strategyName,
+      gatewayCode: gateway_code,
+      number,
+      month,
+      year,
+    };
+
+    const preflightDataCollection = Promise.race([new Promise(resolve => {
+      const listener = ({ data }) => {
+        try {
+          const body = JSON.parse(data);
+          if (body.MessageType === 'profile.completed') {
+            debug('received device data session id', body);
+            resolve({ session_id: body.SessionId });
+            recurly.bus.off('raw-message', listener);
+            frame.destroy();
```

Comment:
This frame reference won't be valid at runtime because they are in different execution scopes.

I think this `Promise.race` will need to be performed in the promise chain begun by `recurly.request.post` below.